prov/efa/src/rxr/rxr_pkt_cmd.c

```diff
@@ -287,9 +287,17 @@ ssize_t rxr_pkt_post_ctrl_once(struct rxr_ep *rxr_ep, int entry_type, void *x_en
 	 * if inject, there will not be completion, therefore tx_pkt_entry has to be
 	 * released here
 	 */
-	if (inject)
+	if (inject) {
 		err = rxr_pkt_entry_inject(rxr_ep, pkt_entry, addr);
-	else if (pkt_entry->send->iov_count > 0)
+		/* Currently only EOR packet is injected using shm ep
+		 * It indicates that the x_entry is rx_entry, and the receiver has
+		 * already received all data using fi_read via shm provider.
+		 * If EAGAIN happens, we should not release rx_entry, as it will
```

Comment:
Why isn't this true for other packet types? It's confusing to have a specific packet type check here.

Reply:
Similarly, the function `rxr_pkt_entry_inject()` is ONLY used by shm EP injecting EOR packet, no other packet type uses it. 
I can move `rxr_release_rx_entry(rxr_ep, x_entry);` inside `rxr_pkt_entry_inject()`, if that's more acceptable.